students/aravichander/lesson07/Mailroom_unittest.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+
+from Mailroom_lesson07 import createlist,getlist,sumdonor
+
+class MyFuncTestCase(unittest.TestCase):
```

Comment:
you are welcome to use unittest if you want -- but it's extra overhead for little gain. That's why we teach and recommend using pytest and simple test functions.

for example, these tests would be as simple as:

```
def test_createlist(self):
    # note -- this kinda shows why you want createlist() to return the list...
    createlist()
    output = getlist()
    output["Bill Gates"] == [10000,12000] 

def test_find_not_there():
    with self.assertRaises(ValueError):
        searchdonors("Carrot Top")
# though I'd probably have searchdonors() return None if it's not there, rather than raising an error.

def test_sumdonor(self):
    # note that you are changing a global with each test run -- not good. which is why you want it to return data, rather than altering a global. This all gets easier with OO -- that's one reason to do it :-)
    createlist()
    sumdonor("Paul Allen") == 1050
```